lib/ansible/modules/windows/win_dns_client.ps1

```diff
@@ -77,35 +79,168 @@ Function Get-NetAdapterInfo {
             $cim_params = @{
                 ClassName = "Win32_NetworkAdapterConfiguration"
                 Filter = "InterfaceIndex = $($_.InterfaceIndex)"
-                Property = "DNSServerSearchOrder", "IPEnabled"
+                Property = "DNSServerSearchOrder", "IPEnabled", "SettingID"
             }
-            $adapter_config = Get-CimInstance @cim_params
+            $adapter_config = Get-CimInstance @cim_params |
+                Add-Member -MemberType AliasProperty -Name InterfaceGuid -Value SettingID -Force -PassThru
+
             if ($adapter_config.IPEnabled -eq $false) {
                 return
             }
 
-            if (Get-Command -Name Get-DnsClientServerAddress -ErrorAction SilentlyContinue) {
-                $dns_servers = Get-DnsClientServerAddress -InterfaceIndex $_.InterfaceIndex | Select-Object -Property @(
-                    "AddressFamily",
-                    "ServerAddresses"
-                )
+            $reg_info = $adapter_config |
+                Get-RegistryNameServerInfo
+
+            [PSCustomObject]@{
+                Name = $_.Name
+                InterfaceIndex = $_.InterfaceIndex
+                InterfaceGuid = $_.InterfaceGuid
+                RegInfo = $reg_info
+            }
+        }
+
+        if (@($net_info).Count -eq 0 -and -not $Name.Contains("*")) {
+            throw "Get-NetAdapterInfo: Failed to find network adapter(s) that are IP enabled with the name '$Name'"
+        }
+
+        $net_info
+    }
+}
+
+Function Get-RegistryNameServerInfo {
+    [CmdletBinding()]
+    Param (
+        [Parameter(ValueFromPipeline=$true,ValueFromPipelineByPropertyName=$true,Mandatory=$true)]
+        [System.Guid]
+        $InterfaceGuid
+    )
+
+    Begin {
+        Set-StrictMode -Off  # Current Scope
```

Comment:
Don't do this, work with strict mode, there's a reason why it is enabled as it makes sure we are more guarded when checking for properties and values.

Reply:
Understood, I didn't do it lightly. This syntax: `$iprop.($v4Items.EnableDhcp)` won't work in strict mode when the item is not on the object (in this case means the key doesn't exist in the registry. It's possible to workaround for sure but checking for a property existing and then doing assignment in all the places it's needed was a lot less readable imo, particularly on combined assignment conditionals like this:
```if (($ns = $iprop.($v4Items.DhcpNameServer)))```
Or bool conversions
```UsingDhcp = $iprop.($v4Items.EnableDhcp) -as [bool]```

I'll work on it; maybe converting the objects to `[hashtable]`s, and then using `.ContainsKey()` before accessing. Or I'll offload to it a function.

Reply:
If you are wanting to check if a property is set on an actual PSObject you can do `$var.PSObject.Properties.Name.Contains('property name')`. Having it as a hashtable would definitely be easier to check if a key exists but either works.

Reply:
I ended up using `Get-AnsibleParam` which was well-positioned to deal with this already. But had to add an additional function to modify the object slightly (see my other comment on that function).

Also went took out the duplicated code per-address family and just made it into a loop.